### PR TITLE
Add menu item to make the Studio to float on top of other windows

### PR DESCRIPTION
**Pull request by sejas**
- Closes https://github.com/Automattic/dotcom-forge/issues/8985

## Proposed Changes

- Added a new menu item to make the Studio App floating over the rest of windows, to make easier to interact with the assistant and the window site.
- This option is not saved as a user preference and it's set as the default false (non floating) after Studio restarts, this respects how QuickTime works.

## Testing Instructions

- Run `npm start`
- Click on View in the the top menu
- Observe the new menu item "Float on Top of All Other Windows"
- Click on the menu item
- Observe the Studio window floats over all other screens
- Click again in the menu item "Float on Top of All Other Windows" to deselect it
- Observe the Studio window does not float, and other windows can be on top of it

https://github.com/user-attachments/assets/a502b71b-ae8a-43e9-9367-56ad8d6b3337

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/menu.ts
Comment (fluiddot) [inline src/menu.ts:155]: I wonder if this item should be under the Window tab instead of View. In fact, I noticed that in the designs it's under that section. WDYT?

Screenshot 2024-09-05 at 12 55 50

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! Tested on macOS and Windows.

I added a comment regarding where to place the new item but shouldn't block the PR.

File: src/menu.ts
Comment (katinthehatsite) [reply src/menu.ts:155]: I had the same thought when testing this +1

**Review by katinthehatsite — APPROVED** [review]
The changes look good and the functionality works as expected. It would be nice to move this to the Window section as suggested by another comment

File: src/menu.ts
Comment (sejas) [reply src/menu.ts:155]: Great Observation,

I've decided to put the menu under View to follow QuickTime settings:

quicktime


I've checked other apps like VLC which displays that menu item under video instead of Window.

vlc

I've also observed that we don't have a "Window" menu item for Windows, we could add it but it would be the only item there.

windows-menu

**Comment by sejas** [comment]
@fluiddot , @katinthehatsite , Thanks for reviewing it. I've updated the code to fix a bug in Windows, I would appreciate a second review.

I decided to put it under View to follow other apps like QuickTime. Let me know what you think. The change is trivial.

While testing on Windows, I've noticed the toggle was not working. Once the user clicked on the menu item, it was not "checked" and the window was floating constantly even after clicking the menu item for second or third time.

Fixed here: https://github.com/Automattic/studio/pull/511/commits/99908d1944205f1a9c573aeab053ec37a3f4607a

File: src/menu.ts
Comment (fluiddot) [reply src/menu.ts:155]: Good point. To be honest, I don't have a strong preference between the View and Window tabs to place the item so looks good to me to put it under View. My comment was more regarding the discrepancy with the design.

**Comment by fluiddot** [comment]
> While testing on Windows, I've noticed the toggle was not working. Once the user clicked on the menu item, it was not "checked" and the window was floating constantly even after clicking the menu item for second or third time.

@sejas Ah, interesting. I didn't encounter this when I tested it on Windows. What Windows version have you tested it?

**Comment by sejas** [comment]
@fluiddot , I'm using Windows 11 Home.

I've observed that the menu in Windows is being "rendered" inside the App, and that's probably why the auto-toggle is not working out of the box.

I encountered a couple of related issues:
- https://github.com/electron/electron/issues/10845
- https://github.com/electron/electron/issues/5055

They solve it by calling `trayIcon.setContextMenu `, but we are not using any [`Tray`](https://github.com/electron/electron/blob/4b5a83197911ce9de80004437d7ff41028e33bcd/docs/api/tray.md) which we don't. My understanding is that the menu is generated each time a user opens it in Windows.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! I've re-tested it in both macOS and Windows and worked nice.

**Pull request merged**
